projects/observability/src/pages/custom-dashboards/custom-dashboard.service.ts

```diff
@@ -0,0 +1,77 @@
+import { HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { UserPreferenceService } from '@hypertrace/common';
+import { PageEvent } from '@hypertrace/components';
+import { Dashboard } from '@hypertrace/hyperdash';
+import { Observable, of, Subscription } from 'rxjs';
+import { GraphQlFilterDataSourceModel } from './../../shared/dashboard/data/graphql/filter/graphql-filter-data-source.model';
+import { GraphQlFilter } from './../../shared/graphql/model/schema/filter/graphql-filter';
+import { PanelData } from './custom-dashboard-store.service';
+
+export interface DashboardListItem {
+  id: string;
+  name: string;
+  panels: PanelData[];
+}
+
+export interface CustomDashboardListResponse {
+  error: object;
+  payload: CustomDashboardPayload[];
+  totalRecords: number;
+  success: boolean;
+}
+interface CustomDashboardResponse {
+  error: object;
+  payload: CustomDashboardPayload;
+  success: boolean;
+}
+export interface CustomDashboardPayload {
+  CreatedAt: Date;
+  Data: DashboardListItem;
+  DeletedAt: Date;
+  Id: string;
+  OwnerID: number;
+  UpdatedAt: Date;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CustomDashboardService {
+  public static readonly API_ID_PARAM_NAME: string = 'dashboard_id';
+  public readonly BASE_URL: string = '/v1/dashboard';
```

Comment:
Why are these public? Are they needed somewhere outside this service?

Reply:
`API_ID_PARAM_NAME` is needed need routing module. Changing BASE_URL to private